client/src/Utils/Contacts/useContactFields.ts

```diff
@@ -55,14 +58,23 @@ const useContactFields = (contactStatus?: InteractedContact['contactStatus']) =>
         : contactType !== ContactType.TIGHT;
 
     const validateContact = (contact: InteractedContact, validationReason: ValidationReason): validValidation | invalidValidation => {
-        if (contact.identificationType === IdentificationTypes.ID && !isIdValid(contact.identificationNumber)) {
-            return { valid: false, error: 'שדה ת.ז. אינו תקין' };
-        };
 
-        if (contact.identificationType === IdentificationTypes.PASSPORT && !isPassportValid(contact.identificationNumber)) {
+        const contactIdType = contact.identificationType.id ? contact.identificationType.id : contact.identificationType;
+        const contactIdNumber = contact.identificationNumber;
+
+        if (contactIdType === identificationTypes[0].id && !isIdValid(contactIdNumber)) {
+            return { valid: false, error: 'שדה ת"ז אינו תקין' };
+        };
+        if (contactIdType === identificationTypes[1].id && !isPassportValid(contactIdNumber)) {
             return { valid: false, error: 'שדה דרכון אינו תקין' };
         };
-
+        if ((contactIdType === identificationTypes[2].id || contactIdType === identificationTypes[3].id ) && !isOtherIdValid(contactIdNumber)) {
+            return { valid: false, error: 'שדה מזהה אינו תקין' };
+        };
+        if (contactIdType === identificationTypes[4].id && !isPalestineIdValid(contactIdNumber)) {
+            return { valid: false, error: 'שדה ת"ז פלסטינית אינו תקין' };
+        };
```

Comment:
consider using a switch statement , or seperate into a function like `validateId(idNum , idType)` with something like an enum to determine the eventual error